src/librustdoc/formats/cache.rs

```diff
@@ -574,20 +574,20 @@ fn add_item_to_search_index(tcx: TyCtxt<'_>, cache: &mut Cache, item: &clean::It
     );
     let aliases = item.attrs.get_doc_aliases();
     let deprecation = item.deprecation(tcx);
-    let index_item = IndexItem {
-        ty: item.type_(),
-        defid: Some(defid),
+    let index_item = IndexItem::new(
```

Comment:
This is not an improvement. The advantage of setting fields is that you actually know what the field name is. In a few cases here, we just have `Some()` or `None`.